src/DGtal/geometry/curves/AlphaThickSegmentComputer.h

```diff
@@ -0,0 +1,700 @@
+ * implementation exploits the height/width maintenance defined from
+ * the \cite FaureTangential2008 (see \cite FaureTangential2008 page
+ * 363) which reduces the complexity from \f$O(n\ log\ n) \f$ into
+ * \f$O( log\ n) \f$.  Note that the convexhull maintenance in linear
+ * time (with point substraction) proposed by Buzer \cite
+ * lilianComputing2007 is not yet implemented.
+ *
+ *
+ *
+ * @tparam TInputPoint the type of the input point which have to be of
+ * dimension 2.
+ * 
+ * @tparam TConstIterator the type of iterator of candidate points
+ * (used in the init with iterator) which should be readable and
+ * forward. By default the iterator is set to the const_iterator of std::vector< TInputPoint > .
+ *
+ *
+ * This class is a model of ForwardContainer and  CForwardSegmentComputer.  
+ * It is also default constructible, copy  constructible, assignable and equality comparable.
+ *
+ *
+ * A typical application of an alpha thick segment recognition can be
+ * done as follows:
+ * - If you consider input point with floating coordinates, you can define this type:
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementNoisyTypedef
+ *
+ * - Then import eventually a vector containing the input points by using the PointListReader class:
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementNoisyReadFile 
+ * 
+ * - Finally apply the segment recognition (here of maximal thickness 15)  by adding the sequence (forward) of contour points: 
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementNoisInitAndReco 
+ *  @note The maximal thickness given in initialization is the same than
+ *  the width given by the method width() of the ParallelStrip primitive
+ *  (not the vertical/horizontal width of the convex hull).
+ *
+ *  
+ * - If you use a Board2D display, you can draw the resulting segment like other 2D objects:
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementDisplay 
+ * 
+ * The complete example of segment recognition is given in exampleAlphaThickSegmentNoisy.cpp
+ *
+ * @note You can also construct the segment by using an input point iterator in initialisation (see @ref moduleAlphaThickSegmentReco for more details) 
+ *
+ * The proposed implementation is mainly a backport from
+ * [ImaGene](https://gforge.liris.cnrs.fr/projects/imagene) with some
+ * various refactoring.
+ */
+
+
+template <typename TInputPoint,
+          typename TConstIterator = typename std::vector< TInputPoint >::const_iterator >
+
+class AlphaThickSegmentComputer
+{
+
+  // ----------------------- public types --------------------------------------
+  BOOST_STATIC_ASSERT(( TInputPoint::dimension == 2 ));
+  
+public:
+  typedef TInputPoint InputPoint;
+  typedef InputPoint InputVector;
+  
+  typedef std::vector< InputPoint > InputPointContainer;
```

Comment:
if the input point container is hardcoded in the class, why do you need a ```const_iterator``` template parameter ?
can ```TConstIterator``` be different from``` InputPointContainer::const_iterator```